src/pybind/mgr/rgw/module.py

```diff
@@ -311,18 +311,22 @@ def rgw_zone_create(self,
                         start_radosgw: Optional[bool] = True,
                         zone_endpoints: Optional[str] = None,
                         inbuf: Optional[str] = None) -> Any:
+
         if inbuf:
             try:
                 rgw_specs = self._parse_rgw_specs(inbuf)
             except RGWSpecParsingError as e:
                 return HandleCommandResult(retval=-errno.EINVAL, stderr=f'{e}')
         elif (zone_name and realm_token):
             token = RealmToken.from_base64_str(realm_token)
-            placement_spec = PlacementSpec.from_string(placement) if placement else None
+            if (type(placement) == dict):
+                placement_spec = PlacementSpec.from_json(placement) if placement else None
+            elif(type(placement) == str):
+                placement_spec = PlacementSpec.from_string(placement) if placement else None
             rgw_specs = [RGWSpec(rgw_realm=token.realm_name,
                                  rgw_zone=zone_name,
                                  rgw_realm_token=realm_token,
-                                 rgw_frontend_port=port,
+                                 rgw_frontend_port=int(port),
```

Comment:
There's no need to enforce the cast here. This method expects the port to be an int (that what the signature and typing enforce). In case you are using a string please parse it in the origin (when you read the int from the dashboard form).

Reply:
Updated the PR